controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,139 @@
+    // Only one supervisor instance can exist in a single controller. As such the same one will be used
+    // every time
+    webots::Supervisor supervisor = webots::Supervisor();
+    for (int i = 2; i < argc; i++) {
+        // Load the Node of the robot by def argument and add to vector
+        otherRobotsNodes.emplace_back(supervisor.getFromDef(argv[i]));
+    }
+
+    // AxisAngle rotations will be read from a config file and saved here
+    std::vector<std::array<double, 4>> rotations;
+    double minDistance;
+    double zHeight;
+    // Load config file
+    try {
+        YAML::Node config = YAML::LoadFile("config.yaml");
+        rotations         = config["rotations"].as<std::vector<std::array<double, 4>>>();
+        minDistance       = config["minDistance"].as<double>();
+        zHeight           = config["zHeight"].as<double>();
+    }
+    catch (const YAML::BadFile& e) {
+        std::cerr << e.msg << std::endl;
+        return 1;
+    }
+    catch (const YAML::ParserException& e) {
+        std::cerr << e.msg << std::endl;
+        return 2;
+    }
+
+    webots::Node* tempNode = supervisor.getFromDef("test");
+    const double xSize     = tempNode->getField("xSize")->getSFFloat();
+    const double ySize     = tempNode->getField("ySize")->getSFFloat();
+
+    // Get the time step of the current world.
+    int timeStep = int(supervisor.getBasicTimeStep());
+
+    // Generate random seed
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+
+    // Generate distributions for random number systems in the following loop
+    std::uniform_real_distribution<> xDistrib(-xSize * 0.5, xSize * 0.5);
+    std::uniform_real_distribution<> yDistrib(-ySize * 0.5, ySize * 0.5);
+    std::uniform_real_distribution<> rotDistrib(0.0, rotations.size() - 1.0);
+
+    while (supervisor.step(timeStep) != -1) {
+
+        // Declare a vector of positions that will be saved as they are randomly generated, to be later
+        // applied to each robot
+        std::vector<std::array<double, 3>> positions = std::vector<std::array<double, 3>>();
+        // Loop through every robot
+        for (size_t i = 0; i < otherRobotsNodes.size(); i++) {
+            // Assume there is no collision
+            bool collision = false;
+            // newPos will be a "Proposed location" for a robot to teleport to
+            std::array<double, 3> newPos;
+            do {
+                collision = false;
+                // Generate a new random location
+                newPos[0] = xDistrib(gen);
+                newPos[1] = yDistrib(gen);
+                newPos[2] = zHeight;
+
+                // Loop through the vector of existing proposed locations and see if the new one is going to
+                // collide with any of them
+                for (std::array<double, 3> testPos : positions) {
```

Comment:
`const` because you aren't modifying the arrays. `auto` because less characters (although this is optional, main benefit IMO is when you either don't know (or care about) the type or when the type is just template messiness). `&` to avoid making copies of the arrays
```suggestion
                for (const auto& testPos : positions) {
```

Reply:
That is definitely much cleaner